test/example/rev.erl

```diff
@@ -0,0 +1,137 @@
+-module(rev).
+-compile(export_all).
+-compile(nowarn_export_all).
+
+-include("../../include/quicer.hrl").
+
+%% This is an axample of how we can revert connections. This is especially
+%% interesting when we want to connect to a device behind a NAT device.
+%% Code behind the NAT device can do connect(), and following that the
+%% code at the "top site" can do start_stream() to the device behind the NAT
+
+%% To run:
+
+%% > Top = rev:start().
+%% > Top ! connect.
+%% > Top ! connect.
+
+
+-define(INTERVAL, 3000).
+
+
+
+start() ->
+    application:ensure_all_started(quicer),
+    _C = proc_lib:spawn_link(fun() ->
+                                     nat_site()
+                             end),
+
+    Top = proc_lib:spawn_link(fun() ->
+                                      listener()
+                              end),
+    Top.
+
+listener() ->
+    Port = 4567,
+    LOptions = [ {cert, "./server.pem"}
+               , {key,  "./server.key"}
+               , {verify, none}
+               , {active, false}
+               , {handshake_idle_timeout_ms, 3 * ?INTERVAL}
+               , {keep_alive_interval_ms, ?INTERVAL}
+               , {alpn, ["sample"]}
+               , {idle_timeout_ms, 3 *?INTERVAL}
+               , {peer_bidi_stream_count, 64000}
+               ],
+
+    {ok, L} = quicer:listen(Port, LOptions),
+    {ok, Conn} = quicer:accept(L, [{active, false}], infinity),
+    {ok, Conn} = quicer:handshake(Conn),
+    r_loop(Conn).
+
+r_loop(Conn) ->
+    receive
+        connect ->
+            io:format("Starting stream downwards \n",[]),
+            {ok, S} = quicer:start_stream(Conn, [{active, false}]),
```

Comment:
This call hides the error when peer is flow controlled on the number of streams. 
you can use  open flag `QUIC_STREAM_START_FLAG_FAIL_BLOCKED` to fail the start instead.
https://github.com/emqx/quic/blob/main/include/quicer_types.hrl#L139

The default behavior is waiting for peer to remove the limit , thus long blocking here.